pages/go/README.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Go testing
+permalink: /go/
+---
+
+## The `testing` package
+
+### What Is It?
+Go's [testing](https://golang.org/pkg/testing/) package provides automated testing out-of-the-box.
+
+### When to Use It?
+Great for:
+
+1. unit testing
+2. benchmarks (i.e. testing how long a certain section of code runs)
+
+### How to Use It?
+
+Here's an example of writing a unit test
+
+#### Sample source file
+Given a source code file `hello.go` which just prints out "Hello" twice via a function `SayTwice` upon execution:
+
+```go
+package main
+
+import (
+	"fmt"
+)
+
+func SayTwice(text string) string {
+	return text + text
+}
+
+func main() {
+	fmt.Println(SayTwice("Hello"))
+}
+```
+
+#### Creating unit tests.
+1. Create a file called `hello_test.go` (all Go test files must have the suffix `_test.go`)
+2. Import `testing`
+3. Idiomatic Go practices tabular testing. This is where there is a single call to the function to be tested and that function is just tested with different inputs and the outputs are tested. For this example, create a simple struct that will hold simple string input and output variables.
```

Comment:
First of all, thanks _soooo_ much for kicking this off, @jcscottiii! This is such a welcome addition!

_But..._

The "tabular testing idiom" is a _horrible_ idea in general. It's suitable for a very, very specific subset of cases. Check out my previous rants on the subject:

* https://pages.18f.gov/automated-testing-playbook/principles-practices-idioms/#avoid-data-driven-tests
* https://mike-bland.com/2013/12/03/go-oh-so-close-to-perfect.html
* https://mike-bland.com/2012/09/11/practice-of-testing.html#testing-how-not-to-write-tests
* https://mike-bland.com/2012/09/11/practice-of-testing.html#testing-concrete-examples

To quote from the lattermost link, I do concede that "if you have a pure function with a few well-defined inputs and outputs that are unlikely to ever change, it can seem like overkill to wrap each one in its own test function." But that's the only time, really. The data-driven way lies madness in every other case.

So what I propose is to break this particular advice out into a new section, and explain what it's good for and what it's not, etc., and perhaps include the first link above for reference.